### ISSUE TITLE
hooks rule

### ISSUE THREAD

**Issue by nandorojo** [intergalacticspacehighway/react-native-popper#2]
https://github.com/intergalacticspacehighway/react-native-popper/blob/70a4ad0b6cc3464e864cb34a1f030b8f7f903196/src/Popover/Popover.tsx#L38

I'm a bit concerned about this; it's possible the trigger could be changed, for instance if you're conditionally rendering it. Is there an alternative option that conditionally rendering a hook?

**Comment by nandorojo** [comment]
You could do this outside of the conditional:

```ts
const triggerExistingProps = React.isValidElement(props.trigger) ? props.trigger.props : {}
const handlers = useOn({
  on: props.on,
  onClose: handleClose,
  onOpen: handleOpen,
  overlayRef,
  ...triggerExistingProps
})
```

And then:

```ts
 else if (React.isValidElement(props.trigger)) { 
    triggerElem = React.cloneElement(props.trigger, {
      ref: triggerRef,
      ...handlers,
    });
  }
```

**Comment by intergalacticspacehighway** [comment]
Thanks! 
This will be resolved in this [PR](https://github.com/intergalacticspacehighway/react-native-popper/pull/3)

**Comment by intergalacticspacehighway** [comment]
Fixed in v0.1.8 🎉
